internal/tenant/tenant.go

```diff
@@ -62,3 +63,42 @@ func (t tenantService) GetTenantInfoByNamespace(apiURL string, ns string) (InfoL
 
 	return tenantInfo, nil
 }
+
+// returns true if the cluster the ns is on has reached maximum capacity
+func (t tenantService) HasReachedMaxCapacity(apiURL, ns string) (bool, error) {
+
+	ti, err := t.GetTenantInfoByNamespace(apiURL, ns)
+	if err != nil {
+		return true, err
+	}
+
+	if len(ti.Errors) != 0 {
+		firstError := ti.Errors[0]
+		err = fmt.Errorf("%s - %s", firstError.Code, firstError.Detail)
+		return true, err
+	}
+
+	if len(ti.Data) == 0 {
+		err = fmt.Errorf("Failed to fetch tenant information for %s", ns)
+		return true, err
+	}
+
+	namespaces := ti.Data[0].Attributes.Namespaces
+	index := indexOfNamespaceWithName(namespaces, ns)
+	if index < 0 {
+		return true, fmt.Errorf("Failed to find %s in tenant info", ns)
+	}
+
+	jenkins := namespaces[index]
+	return jenkins.ClusterCapacityExhausted, nil
+}
+
+// returns the index of the namespace that equals 'name'
+func indexOfNamespaceWithName(namespaces []Namespace, name string) int {
```

Comment:
Why are you returning the `index` instead of the `namespace`.  So you can do something like `if ns == nil {}`

Reply:
1. IndexOf is quiet common and slice[x] is a O(1) operation
2. to return `nil`  the find function would now need to return a pointer to namespace so that the caller can do a `nil` check which I am not a big fan of and doesn't really save us a condition check or anything.

Reply:
I think it is not necessary to return a pointer you can do something like

```
var ns Namespace

for ...... {
  if found {
    ns = namespace
  }
  return ns

}
```

So no pointer. But it was just a comment not from the point of view of performance but as a reader.

Reply:
Have you tried it? my quick trial resulted in a compilation error 

```go
package main

import (
	"fmt"
)

type Person struct {
	name string
	age  int
}

func nilObj() Person {
	var p Person
	return p
}

func main() {
	fmt.Println("%v", nilObj())
	if nilObj() == nil {
		fmt.Println(".... works")
	}
}
```
Am I missing something?

Reply:
Ok true you need a pointer at the end. Nevermind. Humm this golang language :P

Reply:
Actually makes sense to me ... If you go by the spec, a variable declaration will always `0` initialise the variable so, `var x int ` initialises x to 0 and `var s string` to `""` thus avoiding the "million dollar" error of having to do `nil` checks all the time. 

And in my case `struct Person` will be 0 initialised and both name and age will be `"", 0` respectively.